zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingScheduleServiceImpl.java

```diff
@@ -31,10 +33,14 @@ public class ProcessingScheduleServiceImpl
 
   private static final ScheduledTask NOOP_SCHEDULED_TASK = () -> {};
   private static final Logger LOG = Loggers.STREAM_PROCESSING;
+  private static final int PERIODIC_CHECK_INTERVAL_MS = 1000;
+
   private final Supplier<StreamProcessor.Phase> streamProcessorPhaseSupplier;
   private final BooleanSupplier abortCondition;
   private final Supplier<LogStreamWriter> writerSupplier;
   private final StageableScheduledCommandCache commandCache;
+  private final InstantSource clock;
+  private final PriorityQueue<ScheduledTaskImpl> scheduledTasks = new PriorityQueue<>();
```

Comment:
:question: Is this unbounded? If it is, can we either:

1. Bound it. This means somehow handling issues when it _is_ bounded...
2. Have some visibility on its size, e.g. a metric or something. This could help speed up memory issue diagnosis in the future.

Reply:
If we think memory could be an issue, we could try and compare quickly a full `DeadlineTimerWheel` with 1 million entries memory size vs this priority queue. There are some easy utilities that tell you the size of an object (e.g. https://github.com/openjdk/jol/blob/master/README.md)

Reply:
Bounding it would be difficult because nothing is set up to handle failure cases. Just as an example, what would it even mean if a `DueDateChecker` cannot schedule next execution?

For size comparison:
Before this PR, every scheduled timer would end up as an entry in the `DeadlineTimerWheel` (roughly two longs + inevitable overhead) _and_ a `TimerSubscription` on the actor task. Both implementations of `TimerSubscription` require 46 bytes according to JOL.

After this PR, every scheduled timer is a `ScheduledTaskImpl` in the priority queue, each taking 24 bytes according to JOL.

Overall this is a reduction in memory usage, with the same danger of unbounded growth as before.